aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -230,131 +414,129 @@ For more information, see [Scope validation in the Web Host topic](xref:fundamen
 
+The factory method of single service, such as the second argument to [AddSingleton&lt;TService&gt;(IServiceCollection, Func&lt;IServiceProvider,TService&gt;)](/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionserviceextensions.addsingleton#Microsoft_Extensions_DependencyInjection_ServiceCollectionServiceExtensions_AddSingleton__1_Microsoft_Extensions_DependencyInjection_IServiceCollection_System_Func_System_IServiceProvider___0__), doesn't need to be thread-safe. Like a type (`static`) constructor, it's guaranteed to be called once by a single thread.
+
 ## Recommendations
 
 When working with dependency injection, keep the following recommendations in mind:
 
-* DI is for objects that have complex dependencies. Controllers, services, adapters, and repositories are all examples of objects that might be added to DI.
-
-* Avoid storing data and configuration directly in DI. For example, a user's shopping cart shouldn't typically be added to the services container. Configuration should use the [options pattern](xref:fundamentals/configuration/options). Similarly, avoid "data holder" objects that only exist to allow access to some other object. It's better to request the actual item needed via DI, if possible.
+* Avoid storing data and configuration directly in the service container. For example, a user's shopping cart shouldn't typically be added to the service container. Configuration should use the [options pattern](xref:fundamentals/configuration/options). Similarly, avoid "data holder" objects that only exist to allow access to some other object. It's better to request the actual item via dependency injection, if possible.
 
 * Avoid static access to services.
 
-* Avoid service location in your application code.
+* Avoid service location in app code.
 
 * Avoid static access to `HttpContext`.
 
-Like all sets of recommendations, you may encounter situations where ignoring one is required. We have found exceptions to be rare -- mostly very special cases within the framework itself.
+Like all sets of recommendations, you may encounter situations where ignoring a recommendation is required. Exceptions are rare&mdash;mostly special cases within the framework itself.
 
-Dependency injection is an *alternative* to static/global object access patterns. You may not be able to realize the benefits of DI if you mix it with static object access.
+Dependency injection is an *alternative* to static/global object access patterns. You may not be able to realize the benefits of dependency injection if you mix it with static object access.
 
 ## Additional resources
 
 * [Dependency injection into views](xref:mvc/views/dependency-injection)
 * [Dependency injection into controllers](xref:mvc/controllers/dependency-injection)
 * [Dependency injection in requirement handlers](xref:security/authorization/dependencyinjection)
+* [Repository pattern](xref:fundamentals/repository-pattern)
```

Comment:
Prefer the <xref:uid> pattern for these additional resources links.

Reply:
😕 They do. What do you mean?

Reply:
Sorry, my comment didn't display as expected. I meant use `<xref:mvc/views/dependency-injection>` for example. The link text is applied for you.

Reply:
IIRC the decision that was made back in Tom days was to split the "ASP.NET Core" out by specifying the shorter manual titles. Looks like a change of direction was settled upon since then. Are we doing this for all cross-links or just for *Additional resources* section links?